Kudu.Contracts/SiteExtensions/ISiteExtensionManager.cs

```diff
@@ -18,6 +18,11 @@ public interface ISiteExtensionManager
         /// </summary>
         Task<SiteExtensionInfo> InstallExtension(string id, string version, string feedUrl);
 
+        /// <summary>
+        /// Before perform a site extension installation, create setting file to keep track of async operation
+        /// </summary>
+        Task<SiteExtensionInfo> InitInstallSiteExtension(string id);
```

Comment:
agree that it is implementation details.

initially i was thinking to have all the logic that interact with the "SiteExtension.json" setting all inside "SiteExtensionManager.cs", but due to complicate logic, have to work against setting in "GetLocalExtension" , which is in controller.  so i think i can remove this interface method and put the code in controller.